src/WebForms/WebPartsServiceExtensions.cs

```diff
@@ -0,0 +1,197 @@
+// MIT License.
+
+using System.Collections;
+using System.Web.UI;
+using System.Web.UI.WebControls.WebParts;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+
+[assembly: TagPrefix("System.Web.UI.WebControls.WebParts", "asp")]
+
+namespace WebForms;
+
+internal static class WebPartsServiceExtensions
+{
+    internal static IWebFormsBuilder AddWebParts(this IWebFormsBuilder builder)
+    {
+        builder.Services.TryAddSingleton(TimeProvider.System);
+        builder.Services.TryAddSingleton<PersonalizationProvider, InMemoryPersonalizationProvider>();
+        builder.Services.TryAddSingleton<PersonalizationProviderCollection>();
+        builder.Services.AddOptions<WebPartsOptions>()
+            .Configure(options =>
+            {
+                options.AllowedCapabilities.Add(WebPartPersonalization.EnterSharedScopeUserCapability.Name);
+                options.AllowedCapabilities.Add(WebPartPersonalization.ModifyStateUserCapability.Name);
+            });
+
+        return builder;
+    }
+
+    /// <summary>
+    /// A default personalization provider implementation modeled SqlPersonalizationProvider but using an in-memory cache 
```

Comment:
@twsouthwick as a part of webpart do we save state permanently to SQL Server ? If you have plan to implement SQL Server, let me know.. I probably can implement for MySQL to stay truly open source :)

Reply:
I'm starting with an in memory implementation. I have a branch to start migrating some sql stuff, but we'll do that and other db's in a different assembly